Example of folder structure for selection on DeepSportRadar Instance Segmentation Challenge

Ra1Nik opened the pull request:
This is an example of folder structure of a complete run on [DeepSportRadar Instance Segmentation Challenge](https://github.com/DeepSportRadar/instance-segmentation-challenge)!
Predictions are computed on the union of val and test sets using a detectron2 mask_rcnn_R_50_FPN_3x model finetuned on the train set. Metadata are created with the name of the folders in which images are contained.

Commit:
Add example of folder structure of a complete run on DeepSportRadar Instance Segmentation Challenge dataset

IgorSusmelj commented on README.md:
```diff
@@ -1,2 +1,25 @@
-# object_detection_example_structure
-Example of folder structure of a complete run on DeepSportRadar Instance Segmentation Challenge dataset
+### Object Detection example on a selection on SportRadar Challenge
```
Would rename to `Object Detection Example in Lightly Format`

IgorSusmelj commented on README.md:
```diff
@@ -1,2 +1,25 @@
-Example of folder structure of a complete run on DeepSportRadar Instance Segmentation Challenge dataset
+### Object Detection example on a selection on SportRadar Challenge
+
+This is an example of folder structure of a complete run on [DeepSportRadar Instance Segmentation Challenge](https://github.com/DeepSportRadar/instance-segmentation-challenge)!
```
Would change this slightly to:
```
This is an example of folder structure in the Lightly format for data selection using data from the [DeepSportRadar Instance Segmentation Challenge](https://github.com/DeepSportRadar/instance-segmentation-challenge)!
```

IgorSusmelj commented on lightly_bucket/.lightly/metadata/KS-FR-BLOIS/24330/camcourt1_1513714448590_0.json:
```diff
@@ -0,0 +1 @@
+{"file_name": "KS-FR-BLOIS/24330/camcourt1_1513714448590_0.png", "type": "image", "metadata": {"court": "KS-FR-BLOIS", "game_id": "24330"}}
```
When dumping the metadata as json files I suggest using the `indent=4` parameter to format them more nicely.

E.g. 
```python
with open(path_to_prediction, 'w') as f:
    json.dump(lightly_prediction, f, indent=4) 
```

IgorSusmelj commented on lightly_bucket/.lightly/predictions/object_detection_sportradar/schema.json:
```diff
@@ -0,0 +1 @@
+{"task_type": "object-detection", "categories": [{"id": 0, "name": "human", "supercategory": "none"}]}
```
Also here, try to dump this using `indent=4` for nicer/ more human readable formatting :)

IgorSusmelj commented on worker.sh:
```diff
@@ -0,0 +1,2 @@
+#! /usr/bin/bash
+docker run --shm-size="1024m" --gpus all --rm -it     -v OUTPUT_DIR:/home/output_dir lightly/worker:latest token=your_awesome_token worker.worker_id=your_worker_id 
```
Let's stick to the convention in the rest of the examples (e.g. https://docs.lightly.ai/docs/install-lightly#register-the-lightly-worker)
- no output volume mounting
- pass token using environment variable
- use `{}` and capital letters for the placeholders
- use linebreak

All together this would look like this:
```bash
docker run --shm-size="1024m" --gpus all --rm -it -e LIGHTLY_TOKEN={MY_LIGHTLY_TOKEN} \ 
    lightly/worker:latest token=your_awesome_token worker.worker_id={MY_WORKER_ID}
```

IgorSusmelj commented on README.md:
````diff
@@ -1,2 +1,25 @@
+And finally run your Python scheduling script with:
+
+```
+python job_scheduler.py
````
Can we call this script something along the lines of `run_lightly.py`?
(I'd try to use **Run** instead of **Job**)

IgorSusmelj commented on README.md:
````diff
@@ -1,2 +1,25 @@
+Then simply run the worker with your API token and your worker id:
+
+```
+bash worker.sh
````
Let's maybe call this script `start_lightly_worker.sh`?

IgorSusmelj commented on README.md:
```diff
@@ -1,2 +1,25 @@
+
+This is an example of folder structure of a complete run on [DeepSportRadar Instance Segmentation Challenge](https://github.com/DeepSportRadar/instance-segmentation-challenge)!
+Predictions are computed on the union of val and test sets using a detectron2 mask_rcnn_R_50_FPN_3x model finetuned on the train set. Metadata are created with the name of the folders in which images are contained.
+
```
I'd drop here a rough folder structure as a reference such that users get an idea what to expect. You can use tree.

IgorSusmelj reviewed (changes requested):
This looks already very good. I have left a few comments/ suggestions.

Furthermore, please also make use of the relevant filenames feature :)
- e.g. to select a few specific images
- and also to select somewhere a full folder (using `/*`)

Commit:
Add suggested changes

- Fix names
- Make prediction and metadata formats legible
- Add detafolder tree

IgorSusmelj reviewed (approved):
This looks very good! I'd add a small note/sentence in the beginning that makes it clear that this example uses code to work with AWS but the folder structure for other cloud providers stays the same.

Commit:
Add AWS S3 specification

guarin commented on README.md:
````diff
@@ -1,2 +1,60 @@
+
+```
+aws s3 cp input_bucket s3://yourInputBucket/sportradar_input/ --recursive
+aws s3 cp lightly_bucket s3://yourInputBucket/sportradar_lightly/ --recursive
````
I would use `aws s3 cp lightly_bucket s3://yourLightlyBucket/sportradar_lightly/ --recursive` instead of `aws s3 cp lightly_bucket s3://yourInputBucket/sportradar_lightly/ --recursive`. This highlights that two different buckets can be used for input and lightly.

guarin commented on README.md:
````diff
@@ -1,2 +1,60 @@
+aws s3 cp lightly_bucket s3://yourInputBucket/sportradar_lightly/ --recursive
+```
+
+Then simply run the worker with your API token and your worker id:
````
I would drop the "simply" here. It might not be that simple for users. As a user I don't like it if docs say that something is simple and then I struggle with it for a while.

guarin replied:
Could you also add a link to our docs on how to install the lightly worker?

guarin commented on README.md:
````diff
@@ -1,2 +1,60 @@
+
+Then simply run the worker with your API token and your worker id:
+
+```
+bash start_lightly_worker.sh
+```
````
I would copy the docker run command here as it is pretty short.

guarin commented on README.md:
````diff
@@ -1,2 +1,60 @@
+bash start_lightly_worker.sh
+```
+
+And finally run your Python scheduling script with:
+
````
Could you add a note that this requires the lightly python client and link to the docs on how to install it?

guarin commented on start_lightly_worker.sh:
```diff
@@ -0,0 +1,3 @@
+#! /usr/bin/bash
+docker run --shm-size="1024m" --gpus all --rm -it -e LIGHTLY_TOKEN={MY_LIGHTLY_TOKEN} \ 
+    lightly/worker:latest token=your_awesome_token worker.worker_id={MY_WORKER_ID}
```
Could you drop the `token=your_awesome_token` part? The token is already passed with `LIGHTLY_TOKEN={MY_LIGHTLY_TOKEN}`.

guarin commented on README.md:
```diff
@@ -1,2 +1,60 @@
+### Object Detection example on a selection on SportRadar Challenge
+
+This is an example of folder structure in the Lightly format for data selection using data from the [DeepSportRadar Instance Segmentation Challenge](https://github.com/DeepSportRadar/instance-segmentation-challenge)!
+Predictions are computed on the union of val and test sets using a detectron2 mask_rcnn_R_50_FPN_3x model finetuned on the train set. Metadata are created with the name of the folders in which images are contained. KS-FR-ROANNE folder is removed using our [relevant filenames feature](https://docs.lightly.ai/docs/relevant-filenames).
```
I didn't find any mention of `KS-FR-ROANNE` in the relevant filenames file.

Ra1Nik replied:
I canceled the KS-FR-ROANNE from the relevant filename list, which comprehends every filename but the KS-FR-ROANNE ones. Isn't it the proper way to do that?

guarin reviewed (commented):
This looks really nice, great work!

I left a couple comments. We should also check the license whether we are allowed to redistribute the images and if we have to include a license ourselves.

guarin commented on run_lightly.py:
```diff
@@ -0,0 +1,124 @@
+from lightly.openapi_generated.swagger_client.models.datasource_purpose import DatasourcePurpose
+
+# Create the Lightly client to connect to the API.
+client = ApiWorkflowClient(token='your_awesome_token')
```
`client = ApiWorkflowClient(token="MY_LIGHTLY_TOKEN")`

guarin replied:
In general, please follow the same format and naming as in our docs: https://docs.lightly.ai/docs/set-up-your-first-dataset

guarin commented on run_lightly.py:
```diff
@@ -0,0 +1,124 @@
+
+# Input bucket
+client.set_s3_config(
+    resource_path='s3://YourInputBucket/sportradar_input/',
```
Bucket name is not consistent with the aws command in the readme.

guarin commented on run_lightly.py:
```diff
@@ -0,0 +1,124 @@
+
+# Lightly bucket
+client.set_s3_config(
+    resource_path='s3://YourLightlyBucket/sportradar_lightly/',
```
Bucket name is not consistent with the aws command in the readme.

guarin commented on run_lightly.py:
```diff
@@ -0,0 +1,124 @@
+client.schedule_compute_worker_run(
+    worker_config={
+        'enable_corruptness_check': True,
+        'remove_exact_duplicates': True,
```
Please drop `enable_corruptness_check ` and `remove_exact_duplicates `. They are True by default and want to move users away from using those options.

guarin commented on run_lightly.py:
```diff
@@ -0,0 +1,124 @@
+            'batch_size': 16,
+            'shuffle': True,
+            'num_workers': -1,
+            'drop_last': True
```
Not sure if we need the full lightly config here. Maybe only the changed arguments are enough.

Commit:
Fix incosistencies and add links to the docs

Commit:
Modify revelant filenames with folder name
